### PR TITLE
Refactor my-news logic to support grid and list views

### PR DESCRIPTION
**Pull request by phamann**
As discussed new logic
- Both grid and list view and now served via the same route `/myft/my-news`
- If `layout=` query param exists then it takes precedence
- If `myftDashboard` switch is enabled then it `grid` is always shown
- Otherwise users preference is used
- Defaulting to `list` if none of the above applies

I've also refactored the view data model slightly so each layout type can decorate the model with its needs.

Next step is to add the toggling UI.

/cc @Financial-Times/next-myft

### CODE REVIEW

**Comment by adgad** [comment]
LGTM

File: server/controllers/my-news.js [hunk 1]
```diff
@@ -33,24 +33,21 @@ module.exports = function(req, res, next) {
...
-		redirectPromise = topicsPromise
-			.then(function(hasTopics) {
-				shouldShowTour = res.locals.flags.myftProductTour && !hasTopics;
-				return !hasTopics;
-			})
-			.catch(function() {
-				return true;
-			});
+		redirectPromise = myFtApi.getAllRelationship('user', userId, 'followed')
+							.then(() => true, err => {
```
Comment (wheresrhys) [inline server/controllers/my-news.js]: why extra whitespace?

File: server/lib/view-decorators/feed.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (wheresrhys) [inline server/lib/view-decorators/feed.js]: rename this file to fragment?

**Comment by wheresrhys** [comment]
:+1:

**Pull request merged**